test/parallel/test-child-process-spawn-argv0.js

```diff
@@ -0,0 +1,51 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const cp = require('child_process');
+
+
+// Currently, node has no mechanism to actually read argv[0], see 2c6f79c08,
+// https://github.com/nodejs/node/issues/7434
+// https://github.com/nodejs/node/pull/7449
+// https://github.com/nodejs/node/pull/7696
+// for some context on this. As such, we test that setting argv[0] is working
+// by relying on a side-effect in OS X / Linux, namely that the process title
+// will come from the passed in argv[0], and skip this test on platforms where
+// that is not the case.
+if (common.isWindows || common.isSunOS) {
```

Comment:
If the test remains skip()'ed, it might make more sense to turn this into an exclusive list than an inclusive one (I’m pretty sure this will need to be skipped on other OSes, too).